manager/controllers/app/blueprint_controller.go

```diff
@@ -250,6 +258,179 @@ func (r *BlueprintReconciler) applyChartResource(ctx context.Context, cfg *actio
 }
 
+func execPod(k8sClient kubernetes.Interface, config *restclient.Config, podName string, namespace string,
+	command string, stdin io.Reader, stdout io.Writer, stderr io.Writer) error {
+	cmd := []string{
+		"sh",
+		"-c",
+		command,
+	}
+	req := k8sClient.CoreV1().RESTClient().Post().Resource("pods").Name(podName).
+		Namespace(namespace).SubResource("exec")
+	option := &corev1.PodExecOptions{
+		Command: cmd,
+		Stdin:   true,
+		Stdout:  true,
+		Stderr:  true,
+		TTY:     true,
+	}
+	if stdin == nil {
+		option.Stdin = false
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return err
+	}
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  stdin,
+		Stdout: stdout,
+		Stderr: stderr,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type SvcToExpose struct {
+	svcName string
+	port    string
+}
+
+func (r *BlueprintReconciler) getModuleSvcs(ctx context.Context, moduleNamespace string, rel *release.Release) ([]SvcToExpose, error) {
+	var svcsToExpose []SvcToExpose
+	for versionKind := range rel.Info.Resources {
+		for _, obj := range rel.Info.Resources[versionKind] {
+			if unstr, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj); err == nil {
+				res := unstructured.Unstructured{Object: unstr}
+				r.Log.Trace().Msg("resource kind = " + res.GetKind() + " name = " + res.GetName())
+				if res.GetKind() == "Service" {
+					svc := corev1.Service{}
+					svcID := res.GetName()
+					key := types.NamespacedName{Name: svcID}
+					key.Namespace = moduleNamespace
+					if err = r.Get(ctx, key, &svc); err != nil {
+						log.Info().Msgf("Get service returned error. %v", svc)
+					} else {
+						for _, port := range svc.Spec.Ports {
+							r.Log.Trace().Msg("expose service " + svc.GetName() + " with port " + strconv.Itoa(int(port.Port)))
+							svcsToExpose = append(svcsToExpose, SvcToExpose{svc.GetName(), strconv.Itoa(int(port.Port))})
+						}
+					}
+				}
+			} else {
+				log.Err(err).Msg("error getting resources")
+				return nil, err
+			}
+		}
+	}
+	return svcsToExpose, nil
+}
+
+func getSvcHostPort(urlString string) (string, string, string) {
+	servURL, err := managerUtils.ParseRawURL(urlString)
+	if err != nil {
+		log.Err(err).Msgf(CannotParseURLError, urlString)
+		return "", "", ""
+	}
+	hostName := servURL.Hostname()
+	port := servURL.Port()
+	if hostName == "" {
+		log.Warn().Msgf("URL without host name: %s", servURL)
+		return "", "", ""
+	}
+	// Check if it is a local service
+	hostStrings := strings.Split(hostName, ".")
+	if len(hostStrings) > 1 {
+		return hostStrings[0], hostStrings[1], port
+	}
+	return hostStrings[0], "", port
+}
+
+func (r *BlueprintReconciler) applyMBGNetwork(ctx context.Context, blueprint *fapp.Blueprint, network *fapp.ModuleNetwork,
+	rel *release.Release) error {
+	r.Log.Trace().Msg("apply MBG network")
+	fmt.Printf("network is %v\n", network)
```

Comment:
we need to remove fmt.Printf

Reply:
Added just as draft for testing. Will be deleted.